playbooks/openstack/e2e.yaml

```diff
@@ -0,0 +1,206 @@
+      ansible.builtin.slurp:
+        src: "{{ found_files.files[0].path }}"
+      register: metadata_content
+      when: found_files.matched > 0
+    - name: Get cluster-stack and k8s version
+      ansible.builtin.set_fact:
+        cluster_stack_version: "{{ (metadata_content['content'] | b64decode | from_yaml)['versions']['clusterStack'] }}"
+        k8s_version: "{{ (metadata_content['content'] | b64decode | from_yaml)['versions']['kubernetes'] }}"
+      when: metadata_content is defined
+    - name: Parse k8s version to major.minor
+      ansible.builtin.set_fact:
+        k8s_version_major_minor: "{{ k8s_version | regex_replace('^v?([0-9]+\\.[0-9]+)\\..*', '\\1') }}"
+      when: k8s_version is defined
+    - name: Extract cloud name from clouds_yaml_full
+      ansible.builtin.set_fact:
+        cloud_name: "{{ clouds_yaml_full.clouds.keys() | first }}"
+      when: clouds_yaml_full.clouds is defined and clouds_yaml_full.clouds | dict2items | length == 1
+    - name: Generate clusterstack YAML
+      ansible.builtin.template:
+        src: "cluster-stack-template.yaml.j2"
+        dest: "{{ ansible_user_dir }}/clusterstack.yaml"
+        mode: "0644"
+    - name: Apply cluster-stack template
+      ansible.builtin.command: "kubectl apply -f {{ ansible_user_dir }}/clusterstack.yaml"
+      changed_when: true
+    - name: Necessary pause for the clusterstack resource to exist (default is 2 minutes)
+      ansible.builtin.pause:
+        seconds: "{{ wait_for_cluster_stack_resource }}"
+    - name: Wait for cluster-stack to be ready
+      ansible.builtin.command: "kubectl wait clusterstack/clusterstack --for=condition=Ready --timeout={{ wait_for_cluster_stack }}s"
+      changed_when: true
+    - name: Create k8s workload cluster and execute checks
+      block:
+        - name: Create Server Groups for nodes when scs_compliance tests are enabled
+          ansible.builtin.import_tasks: ../tasks/create_server_groups.yaml
+          when: scs_compliance.enabled
+        - name: Generate cluster YAML
+          ansible.builtin.template:
+            src: "cluster.yaml.j2"
+            dest: "{{ ansible_user_dir }}/cluster.yaml"
+            mode: "0644"
+          vars:
+            worker_server_group_id: "{{ srvgrp_worker.stdout | default('') }}"
+            worker_server_group_id_value: "{% if worker_server_group_id == '' %}\"\"{% else %}{{ worker_server_group_id }}{% endif %}"
+            controller_server_group_id: "{{ srvgrp_controller.stdout | default('') }}"
+            controller_server_group_id_value: "{% if controller_server_group_id == '' %}\"\"{% else %}{{ controller_server_group_id }}{% endif %}"
+        - name: Apply cluster template
+          ansible.builtin.command: "kubectl apply -f {{ ansible_user_dir }}/cluster.yaml"
+          changed_when: true
+        - name: Get kubeadmcontrolplane name
+          ansible.builtin.command: "kubectl get kubeadmcontrolplane -o=jsonpath='{.items[0].metadata.name}'"
```

Comment:
When we apply the cluster template, kubeadmcontrolplane doesn't need to be created immediately, and `items` array is empty, so I think we need some retries here, see https://github.com/SovereignCloudStack/cluster-stacks/pull/53#issuecomment-2140079661.

Reply:
I have not yet encountered a case when the `kubeadmcontrolplane` was not created immediately on my local machine when the cluster-stack was working. I think it's more of a slow infrastructure/machine problem where we run this pipeline. But yes, some retries may work, we can try that.

Reply:
see 73e54297e1eaa742e1d5bc122cd8d6a8dfb0becf